### DISCUSSION TITLE
restarting kibana in a kubernetes pod

### DISCUSSION THREAD

**Discussion by ovidiaconescu** [fbaligand/kibana-enhanced-table#223] [Q&A] [answered]
Hey, what is the recommended way to restart kibana within a kubernetes pod after installing the plugin?

If I try to `kill` the process, the entire pod gets restarted.

**Comment by fbaligand** [comment, marked as answer]
Well, globally, on kubernetes, it is really not recommended to modify a pod at runtime, and worst to restart a pod after then. 

So the recommended way is to create a new Docker image that inherits from Kibana image, then installs the plugin. 
Then you run in Kubernetes this new image.

**Reply by fbaligand** [reply]
You can find here a Dockerfile sample:
https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-kibana-plugins.html

**Reply by ovidiaconescu** [reply]
@fbaligand that's exactly the way to go. Custom image worked perfectly

```
FROM docker.elastic.co/kibana/kibana:7.11.2
RUN /usr/share/kibana/bin/kibana-plugin install https://github.com/fbaligand/kibana-enhanced-table/releases/download/v1.11.2/enhanced-table-1.11.2_7.11.2.zip
RUN /usr/share/kibana/bin/kibana --optimize
```
